test/integration/job/job_test.go

```diff
@@ -58,6 +58,309 @@ import (
+			}
+
+			validateJobPodsStatus(ctx, t, clientSet, jobObj, podsByStatus{
+				Active: 1,
+				Failed: 0,
+				Ready:  pointer.Int32(0),
+			}, wFinalizers)
+
+			if err := setJobPodsPhase(ctx, clientSet, jobObj, v1.PodSucceeded, 1); err != nil {
+				t.Fatalf("Failed setting phase %s on Job Pod: %v", v1.PodSucceeded, err)
+			}
+			validateJobSucceeded(ctx, t, clientSet, jobObj)
+			validateFinishedPodsNoFinalizer(ctx, t, clientSet, jobObj)
+		})
+	}
+}
+
+func createJobWithPodFailureMatchingIgnoreRuleOnExitCodes(t *testing.T, ctx context.Context, clientSet clientset.Interface, ns *v1.Namespace, wFinalizers bool) (*batchv1.Job, error) {
+	t.Helper()
+	return createJobWithSinglePod(t, ctx, clientSet, ns, wFinalizers, batchv1.Job{
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32(0),
+			PodFailurePolicy: &batchv1.PodFailurePolicy{
+				Rules: []batchv1.PodFailurePolicyRule{
+					{
+						Action: batchv1.PodFailurePolicyActionIgnore,
+						OnExitCodes: &batchv1.PodFailurePolicyOnExitCodesRequirement{
+							Operator: batchv1.PodFailurePolicyOnExitCodesOpIn,
+							Values:   []int32{1, 2, 3},
+						},
+					},
+				},
+			},
+		},
+	}, v1.PodStatus{
+		Phase: v1.PodFailed,
+		ContainerStatuses: []v1.ContainerStatus{
+			{
+				Name: "main-container",
+				State: v1.ContainerState{
+					Terminated: &v1.ContainerStateTerminated{
+						ExitCode: 2,
+					},
+				},
+			},
+		},
+	})
+
+}
+
+func createJobWithPodFailureMatchingIgnoreRuleOnPodConditions(t *testing.T, ctx context.Context, clientSet clientset.Interface, ns *v1.Namespace, wFinalizers bool) (*batchv1.Job, error) {
+	t.Helper()
+	return createJobWithSinglePod(t, ctx, clientSet, ns, wFinalizers, batchv1.Job{
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32(0),
+			PodFailurePolicy: &batchv1.PodFailurePolicy{
+				Rules: []batchv1.PodFailurePolicyRule{
+					{
+						Action: batchv1.PodFailurePolicyActionIgnore,
+						OnPodConditions: &batchv1.PodFailurePolicyOnPodConditionsRequirement{
+							Operator: batchv1.PodFailurePolicyOnPodConditionsOpIn,
+							Values:   []v1.PodConditionType{v1.PodConditionType("EvictedByAPI")},
+						},
+					},
+				},
+			},
+		},
+	}, v1.PodStatus{
+		Phase: v1.PodFailed,
+		Conditions: []v1.PodCondition{
+			{
+				Type:   v1.PodConditionType("EvictedByAPI"),
+				Status: v1.ConditionTrue,
+			},
+		},
+	})
+}
+
+func createJobWithSinglePod(t *testing.T, ctx context.Context, clientSet clientset.Interface, ns *v1.Namespace, wFinalizers bool, job batchv1.Job, podStatus v1.PodStatus) (*batchv1.Job, error) {
+	t.Helper()
+	jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &job)
+	if err != nil {
+		return nil, err
+	}
+	validateJobPodsStatus(ctx, t, clientSet, jobObj, podsByStatus{
+		Active: 1,
+		Ready:  pointer.Int32(0),
+	}, wFinalizers)
+
+	op := func(p *v1.Pod) bool {
+		p.Status = podStatus
+		return true
+	}
+
+	if err := updateJobPodsStatus(ctx, clientSet, jobObj, op, 1); err != nil {
```

Comment:
This update is too hidden.

I would call it from each `Test` function as necessary.

Reply:
Created a single test table with 3 cases, the update is now called directly from the Test function